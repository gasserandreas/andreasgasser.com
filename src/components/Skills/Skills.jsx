import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Box, Heading } from 'rebass/styled-components';

const Styles = {
  Skills: styled(Box)(({ theme }) => `
    color: ${theme.colors.muted}
  `),
  Title: styled(Heading)(({ theme }) => `
    margin-bottom: 1rem;
    color: ${theme.colors.text};
    text-transform: inherit;
    font-size: 1.2rem;
  `),
};

const Skills = ({ children, title, ...props }) => (
  <Styles.Skills {...props}>
    <Styles.Title variant="section">
      {title}
    </Styles.Title>
    {children}
  </Styles.Skills>
);

Skills.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
};

export default Skills;
