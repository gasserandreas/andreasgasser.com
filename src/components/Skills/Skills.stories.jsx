import React from 'react';
import { Flex } from 'rebass/styled-components';

import Skills from './Skills';

export default {
  title: 'Skills',
  component: Skills,
};

export const defaultSkills = () => (
  <Flex>
    <Skills
      title="Software Development"
      icon={null}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
      {' '}
      ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
      {' '}
      ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </Skills>
  </Flex>
);

defaultSkills.story = {
  name: 'default',
};
