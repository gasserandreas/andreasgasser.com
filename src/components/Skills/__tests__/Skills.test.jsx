/* global withTheme */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Skills from '../Skills';

describe('components/Skills', () => {
  it('simple snapshot test', () => {
    const dom = mount(withTheme(<Skills title="Title">some content</Skills>));
    expect(toJson(dom)).toMatchSnapshot();
  });
});
