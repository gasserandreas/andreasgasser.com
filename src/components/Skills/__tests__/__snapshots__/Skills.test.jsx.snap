// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Skills simple snapshot test 1`] = `
<CustomThemeProvider
  dark={false}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "320px",
          "540px",
          "720px",
          "960px",
          "1140px",
          "1400px",
        ],
        "buttons": Object {
          "link": Object {
            "&:disabled": Object {
              "backgroundColor": "inherit",
              "color": "text",
              "cursor": "not-allowed",
            },
            "&:hover": Object {
              "color": "muted",
              "cursor": "pointer",
            },
            "backgroundColor": "inherit",
            "border": "none",
            "color": "text",
            "fontSize": "0.95rem",
            "fontWeight": 400,
            "height": "2.5rem",
            "padding": "0rem 0.5rem",
            "textTransform": "uppercase",
          },
          "outline": Object {
            "&:disabled": Object {
              "backgroundColor": "inherit",
              "color": "text",
              "cursor": "not-allowed",
            },
            "&:hover": Object {
              "backgroundColor": "text",
              "color": "textInverse",
              "cursor": "pointer",
            },
            "backgroundColor": "inherit",
            "border": "1px solid #000",
            "borderColor": "text",
            "color": "text",
            "fontSize": "0.95rem",
            "fontWeight": 400,
            "height": "2.5rem",
            "padding": "0rem 0.5rem",
            "textTransform": "uppercase",
          },
        },
        "colors": Object {
          "background": "#c3c3c3",
          "backgroundInverse": "#fff",
          "error": "#E11531",
          "muted": "#595959",
          "text": "#333333",
          "textHighlighed": "#333",
          "textHover": "#7e7e7e",
          "textInverse": "#f6f6f6",
        },
        "fontSizes": Array [
          14,
          16,
          20,
          24,
          28,
          32,
          40,
          48,
          56,
          64,
        ],
        "fontWeights": Object {
          "body": 400,
          "bold": 900,
          "heading": 600,
        },
        "fonts": Object {
          "body": "Open Sans, Helvetica Neue, Helvetica, sans-serif",
          "heading": "inherit",
          "monospace": "Menlo, monospace",
        },
        "lineHeights": Object {
          "body": 1.5,
          "heading": 1.125,
        },
        "radii": Object {
          "default": "2px",
        },
        "shadows": Object {
          "card": "0 0 4px rgba(0, 0, 0, .125)",
        },
        "sizes": Object {
          "avatar": 48,
        },
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
        "styles": Object {
          "root": Object {
            "fontFamily": "body",
            "fontWeight": "body",
            "lineHeight": "body",
          },
        },
        "text": Object {
          "bold": Object {
            "color": "#333333",
            "fontSize": Array [
              1,
              1,
              1,
              2,
              2,
              2,
            ],
            "fontWeight": "900",
          },
          "default": Object {
            "color": "#333333",
            "fontSize": Array [
              1,
              1,
              1,
              2,
              2,
              2,
            ],
          },
          "muted": Object {
            "color": "#595959",
            "fontSize": Array [
              1,
              1,
              1,
              2,
              2,
              2,
            ],
          },
          "page": Object {
            "color": "#333333",
            "fontSize": Array [
              4,
              4,
              4,
              5,
              5,
              5,
            ],
            "fontWeight": 600,
            "letterSpacing": "0.075rem",
            "lineHeight": "2.75rem",
            "margin": "2rem 0 2.5rem",
            "maxWidth": "32rem",
            "wordSpacing": "-0.08em",
          },
          "section": Object {
            "color": "#f6f6f6",
            "fontSize": Array [
              1,
              1,
              1,
              2,
              2,
              2,
            ],
            "fontWeight": 900,
            "letterSpacing": "2px",
            "lineHeight": "1.7rem",
            "margin": "1.5rem 0",
            "textTransform": "uppercase",
          },
          "small": Object {
            "color": "#333333",
            "fontSize": Array [
              0,
              0,
              0,
              1,
              1,
              1,
            ],
          },
          "subPage": Object {
            "color": "#595959",
            "fontSize": Array [
              2,
              2,
              2,
              3,
              3,
              3,
            ],
            "fontWeight": 500,
            "letterSpacing": "0.0625rem",
            "lineHeight": "2.2rem",
            "margin": "1.25rem 0 1.75rem",
            "marginBottom": 2,
            "maxWidth": "32rem",
          },
        },
        "variants": Object {
          "avatar": Object {
            "borderRadius": "circle",
            "height": "avatar",
            "width": "avatar",
          },
          "card": Object {
            "bg": "background",
            "boxShadow": "card",
            "p": 2,
          },
          "link": Object {
            "color": "primary",
          },
          "nav": Object {
            ":hover,:focus,.active": Object {
              "color": "primary",
            },
            "color": "inherit",
            "display": "inline-block",
            "fontSize": 1,
            "fontWeight": "bold",
            "p": 2,
            "textDecoration": "none",
          },
        },
      }
    }
  >
    <Skills
      title="Title"
    >
      <Styled(styled.div)>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": false,
                "lastClassName": "jozcqn",
                "rules": Array [
                  "box-sizing: border-box;",
                  "margin: 0;",
                  "min-width: 0;",
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
              },
              "displayName": "Styled(styled.div)",
              "foldedComponentIds": Array [
                "sc-bdVaJa",
              ],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="sc-bdVaJa sc-htpNat jozcqn"
          >
            <Styled(Component)
              variant="section"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bxivhb",
                      "isStatic": false,
                      "lastClassName": "jHURce",
                      "rules": Array [
                        [Function],
                      ],
                    },
                    "displayName": "Styled(Component)",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bxivhb",
                    "target": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                variant="section"
              >
                <ForwardRef
                  className="sc-bxivhb jHURce"
                  variant="section"
                >
                  <styled.div
                    __css={
                      Object {
                        "fontFamily": "heading",
                        "fontSize": 4,
                        "fontWeight": "heading",
                        "lineHeight": "heading",
                      }
                    }
                    as="h2"
                    className="sc-bxivhb jHURce"
                    tx="text"
                    variant="section"
                  >
                    <StyledComponent
                      __css={
                        Object {
                          "fontFamily": "heading",
                          "fontSize": 4,
                          "fontWeight": "heading",
                          "lineHeight": "heading",
                        }
                      }
                      as="h2"
                      className="sc-bxivhb jHURce"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bdVaJa",
                            "isStatic": false,
                            "lastClassName": "hzUygp",
                            "rules": Array [
                              "box-sizing: border-box;",
                              "margin: 0;",
                              "min-width: 0;",
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                              [Function],
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bdVaJa",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      tx="text"
                      variant="section"
                    >
                      <h2
                        className="sc-bdVaJa hzUygp sc-bxivhb jHURce"
                      >
                        Title
                      </h2>
                    </StyledComponent>
                  </styled.div>
                </ForwardRef>
              </StyledComponent>
            </Styled(Component)>
            some content
          </div>
        </StyledComponent>
      </Styled(styled.div)>
    </Skills>
  </ThemeProvider>
</CustomThemeProvider>
`;
